pkg/runner/runner_implementation.go

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright 2025 The Protobom Authors
+
+package runner
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types/ref"
+)
+
+type RunnerImplementation interface {
+	ReadStream(io.Reader) (string, error)
+	Compile(*cel.Env, string) (*cel.Ast, error)
+	Evaluate(*cel.Env, *cel.Ast, *map[string]any) (ref.Val, error)
+}
+
+type defaultRunnerImplementation struct{}
+
+func (dri *defaultRunnerImplementation) ReadStream(reader io.Reader) (string, error) {
+	// Read all the stream into a string
+	contents, err := io.ReadAll(reader)
+	if err != nil {
+		return "", fmt.Errorf("reading stram code: %w", err)
+	}
+	return string(contents), nil
```

Comment:
I saw you had a blank line before the return in `Evaluate` below. If you want, it can be checked by enabling the `nlreturn` linter in `.golangci.yml`

```suggestion
	}

	return string(contents), nil
```